string_manipulation.rb

```diff
@@ -1,30 +1,97 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = my_string.length - 1
+  while i < j do
+    temp = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = temp
+    i += 1
+    j -=1
+  end
+  return my_string

```

Comment:
Since these methods are all doing operations in place, you don't need to return the strings.